src/Cassandra/Client.php

```diff
@@ -41,13 +52,52 @@ class Client implements Session
      */
     public function __construct(array $config)
     {
-        $this->buildCluster($config);
-
+        $this->config = $config;
         $this->session = null;
         $this->keyspace = $config['keyspace'];
     }
 
     /**
+     * Set event dispatcher
+     *
+     * @param EventDispatcherInterface $eventDispatcher
+     */
+    public function setEventDispatcher(EventDispatcherInterface $eventDispatcher)
+    {
+        if ($eventDispatcher instanceof EventDispatcherInterface) {
```

Comment:
The type is already checked by the method signature, we could remove this test to simplify code.